### PR TITLE
Fixed search indexing

### LINKED ISSUE #1194: Rebuilding search index - error if search.field is inside a content block
After I tried to rebuild the search index with following command:
```
app/console massive:search:index:rebuild
```

I've received this error multiple times:

```
Error indexing or de-indexing page (path: /cmf/webspace/contents/page, locale: de): Notice: Array to string conversion
```

Seems like the error comes from a property which has following tag but is **inside a block**:
```
<tag name="sulu.search.field" role="image"/>
```

After removing these lines, it was working again.

### PR DESCRIPTION
**Pull request by dantleech**
This PR fixes the search indexing, which is currently broken with the ZendLucene implementation.

- [x] Test coverage

Fixes #1194

### CODE REVIEW

File: src/Sulu/Bundle/ContentBundle/Search/Metadata/StructureDriver.php [hunk 1]
```diff
@@ -146,7 +146,7 @@ public function loadMetadataForClass(\ReflectionClass $class)
                 array(
                     'type' => 'string',
                     'field' => $this->factory->createMetadataField($prop->getName()),
-                    'aggregated' => true,
+                    'aggregate' => true,
```
Comment (dantleech) [inline src/Sulu/Bundle/ContentBundle/Search/Metadata/StructureDriver.php]: This wasn't indexing at all..

File: src/Sulu/Bundle/ContentBundle/Search/Metadata/StructureDriver.php [hunk 2]
```diff
@@ -188,53 +188,59 @@ public function getAllClassNames()
...
-                                $tagAttributes['role'],
-                                implode(', ', array('title', 'description', 'image'))
-                            )
-                        );
-                }
-            } elseif (!isset($tagAttributes['index']) || $tagAttributes['index'] !== 'false') {
-                $metadata->addFieldMapping(
-                    $property->getName(),
-                    array(
-                        'type' => isset($tagAttributes['type']) ? $tagAttributes['type'] : 'string',
```
Comment (dantleech) [inline src/Sulu/Bundle/ContentBundle/Search/Metadata/StructureDriver.php]: Not much change here, but changed to return early.

**Comment by dantleech** [comment]
Not sure why the tests are passing, will need to investigate.

File: src/Sulu/Bundle/ContentBundle/Search/Metadata/StructureDriver.php [hunk 3]
```diff
@@ -188,53 +188,66 @@ public function getAllClassNames()
...
+                            $tagAttributes['role'],
+                            implode(', ', array('title', 'description', 'image'))
+                        )
+                    );
             }
+
+            return;
+        } 
+
+        if (isset($tagAttributes['role']) && null !== $tagAttributes['role']) {
```
Comment (dantleech) [inline src/Sulu/Bundle/ContentBundle/Search/Metadata/StructureDriver.php]: Add a test.

**Comment by danrot** [comment]
@dantleech Can you use our PR template, and mark the corresponding issue as fixed?

**Comment by dantleech** [comment]
Changed the content tests to use Zend Search instead of the test adapter, as the test adapter was unaware of the `aggregate` feature which was broken here.

I would make this a hotfix, as currently indexing shouldn't work at all.

**Comment by dantleech** [comment]
I guess this is no longer required.

**Pull request closed without merging**